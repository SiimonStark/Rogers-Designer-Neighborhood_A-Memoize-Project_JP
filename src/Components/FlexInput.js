import React, { Component } from 'react';

class FlexInput extends Component {
  constructor() {
    super()
    this.state = {
      flexField0: {flexRule: '', flexValue: ''},
      inputCounter: 0,
      inputMapList: [0]
    }
  }
  handleFlexChange = (e) => {
    const {value, id, className} = e.target
    let flexRule = this.state[id].flexRule
    let flexValue = this.state[id].flexValue;

    className === 'flexRule'
      ? flexRule = value
      : flexValue = value

    this.setState({[id]: {flexRule, flexValue}});
  }
  addInputField = () => {
    let counter = this.state.inputCounter + 1;
    let list = this.state.inputMapList;
    let fields = `flexField${counter}`;
    let newInput = {flexRule: '', flexValue: ''};

    list.push(counter)

    this.setState({inputCounter: counter, inputMapList: list, [fields]: newInput});
  }
  render() {
    console.log('FlexState: ', this.state)
    return (
      <form className="FlexInput">
        <h4>Flex-Input:</h4>
        {this.state.inputMapList.map((inputField, ind) => { return (
          <div className={'flexInput inputGroup' + ind}>
            <input id={'flexField' + ind} onChange={this.handleFlexChange}  className="flexRule" type="text" placeholder="Rule" />
            <span className="colon">:</span>
            <input id={'flexField' + ind} onChange={this.handleFlexChange}  className="flexValue" type="text" placeholder="Value" />
          </div>
        )})}
        <input id="addInput" onClick={this.addInputField} value="+" type="button" />
      </form>
    );
  }
}

export default FlexInput;