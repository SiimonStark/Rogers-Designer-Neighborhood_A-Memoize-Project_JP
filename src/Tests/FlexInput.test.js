import React from "react";
import FlexInput from "../Components/FlexInput";
import { shallow } from "enzyme";

let updateInputsMock = jest.fn();
let mockState = {
  flexField0: { flexRule: "", flexValue: "" },
  inputCounter: 0,
  inputMapList: [0]
};

describe('FlexInput', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FlexInput updateInputs={updateInputsMock} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(mockState);
  });

  it.skip('when input is updated, it should invoke handleFlexChange()', () => {
    const mockEvent = { target: { value: "d" } };
    wrapper.instance().handleFlexChange(mockEvent);
  });

  it('should addInput field onClick of "+" button', () => {
    expect(wrapper.state("inputCounter")).toEqual(0);

    wrapper.find("#addInput").simulate("click");
    wrapper.find("#addInput").simulate("click");
    
    expect(wrapper.state("inputCounter")).toEqual(2);
    expect(wrapper.state("inputMapList").length).toEqual(3);
    expect(wrapper.find("div").length).toEqual(3);
    expect(wrapper).toMatchSnapshot();
  });
});