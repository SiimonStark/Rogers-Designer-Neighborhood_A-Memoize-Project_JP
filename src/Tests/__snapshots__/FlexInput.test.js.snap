// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexInput should addInput field onClick of "+" button 1`] = `
<form
  className="FlexInput"
>
  <h4>
    Flex-Input:
  </h4>
  <div
    className="flexInput inputGroup0"
  >
    <input
      className="flexRule"
      id="flexField0"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="flexValue"
      id="flexField0"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <div
    className="flexInput inputGroup1"
  >
    <input
      className="flexRule"
      id="flexField1"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="flexValue"
      id="flexField1"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <div
    className="flexInput inputGroup2"
  >
    <input
      className="flexRule"
      id="flexField2"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="flexValue"
      id="flexField2"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <input
    id="addInput"
    onClick={[Function]}
    type="button"
    value="+"
  />
</form>
`;

exports[`FlexInput should match snapshot 1`] = `
<form
  className="FlexInput"
>
  <h4>
    Flex-Input:
  </h4>
  <div
    className="flexInput inputGroup0"
  >
    <input
      className="flexRule"
      id="flexField0"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="flexValue"
      id="flexField0"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <input
    id="addInput"
    onClick={[Function]}
    type="button"
    value="+"
  />
</form>
`;
